### PR TITLE
feat: `cds.assert`

### PR DESCRIPTION
**Pull request by SamuelBrucksch**

### CODE REVIEW

**Review by schwma — CHANGES_REQUESTED** [review]

File: test/tests/error-handling-prod.test.js [hunk 1]
```diff
@@ -43,15 +45,22 @@ describe('graphql - error handling in production', () => {
...
-        element: 'notEmptyI',
-        entity: 'ValidationErrorsService.A',
-        message: 'Value is required',
-        numericSeverity: 4,
-        target: 'notEmptyI',
-        type: 'cds.Integer'
-      })
+
+      if (cds.env.features.cds_assert) {
+        // REVISIT: message missing, cds.assert also misses some properties like type, element, entity ...
```
[the remark this answers is not in the document]
Comment (sjvans) [reply test/tests/error-handling-prod.test.js:50]: see https://github.com/cap-js/graphql/pull/162

File: test/tests/error-handling-prod.test.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// process.env.CDS_FEATURES_CDS__ASSERT = 'true'
+
```
Comment (SamuelBrucksch) [inline test/tests/error-handling-prod.test.js:2]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Support for `cds.assert` via `cds.env.features.cds_assert = true` flag
+
```
Comment (SamuelBrucksch) [inline CHANGELOG.md:13]:
```suggestion
```

File: lib/resolvers/crud/update.js [hunk 4]
```diff
@@ -24,6 +24,16 @@ module.exports = async ({ req, res }, service, entity, selection) => {
   const entries = entriesStructureToEntityStructure(service, entity, astToEntries(input))
   query.with(entries)
 
+  // REVISIT: should cds.assert be enabled by default?
```
Comment (SamuelBrucksch) [inline lib/resolvers/crud/update.js:27]:
```suggestion
```

File: lib/resolvers/crud/create.js [hunk 5]
```diff
@@ -13,6 +13,16 @@ module.exports = async ({ req, res }, service, entity, selection) => {
   const entries = entriesStructureToEntityStructure(service, entity, astToEntries(input))
   query.entries(entries)
 
+  // REVISIT: should cds.assert be enabled by default?
```
Comment (SamuelBrucksch) [inline lib/resolvers/crud/create.js:16]:
```suggestion
```

**Review by schwma — APPROVED** [review]

**Comment by schwma** [comment]
Can we merge this?

**Pull request merged**